src/instruments/src/EFB/Utils/routing.tsx

```diff
@@ -31,15 +31,21 @@ type HistoryEntry = {pathname: string, search: string, hash: string, state: any,
 export const findLatestSeenPathname = (history: any, basePath: string): string | undefined => {
     // @ts-ignore
     const historyEntries: HistoryEntry[] = history.entries;
-
     const lastSeenPathname = [...historyEntries].reverse().find(({ pathname }) => pathname.includes(`${basePath}/`))?.pathname;
 
     return lastSeenPathname;
 };
 
 export const PageRedirect = ({ basePath, tabs }: PageRouteProps) => {
     const history = useHistory();
-    const redirectPathname = useRef((() => findLatestSeenPathname(history, basePath) ?? `${basePath}/${pathify(tabs[0].name)}`)());
+    const getRedirectPathname = useCallback(() => findLatestSeenPathname(history, basePath) ?? `${basePath}/${pathify(tabs[0].name)}`, [history, basePath, tabs]);
+    const redirectPathname = useRef(getRedirectPathname());
+
+    useEffect(() => {
+        history.listen(() => {
+            redirectPathname.current = getRedirectPathname();
+        });
```

Comment:
It is good practice to unlisten any listeners added in useEffect when the component unmounts. Otherwise you may be leaking listeners. You can do this by returning a cleanup function inside useEffect.

Reply:
Sorry about that; I forgot.